internal/provider/incident_user_data_source.go

```diff
@@ -0,0 +1,207 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+	"github.com/samber/lo"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentUserDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentUserDataSource{}
+)
+
+func NewIncidentUserDataSource() datasource.DataSource {
+	return &IncidentUserDataSource{}
+}
+
+type IncidentUserDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentUserDataSourceModel struct {
+	BaseRole    *RBACRole    `tfsdk:"base_role" json:"base_role"`
```

Comment:
Do we care about roles here? If not, should we just slim this down to what we actually need to make our life simpler?

Reply:
I think it's much easier if this is a faithful representation of what the API has in it

Reply:
But as soon as we add things to the API, it won't be, right?

Reply:
Like, we can always add these roles later if someone needs them, but all we're doing by including them is making it harder for us to change the representation of the roles later, I think (e.g. changing role slugs)